src/libp2p.js

```diff
@@ -205,107 +43,33 @@ class DaemonLibp2p extends Libp2p {
-    this._peerRouting = new PeerRouting(routing)
-  }
-
-  get dht () {
-    return this._kadDHT
-  }
-
-  set dht (_) {
-    this._kadDHT = new DHT(this)
-  }
-
   /**
    * Starts the libp2p node
-   * NOTE: This is currently promisified internally by libp2p
-   *
-   * @param {function(Error)} callback
-   */
-  start (callback) {
-    super.start((err) => {
-      if (err) return callback(err)
-
-      // replace with announce addrs until libp2p supports this directly
-      if (this.announceAddrs.length > 0) {
-        this.peerInfo.multiaddrs.clear()
-        this.announceAddrs.forEach(addr => {
-          this.peerInfo.multiaddrs.add(addr)
-        })
-      }
-
-      // temporary removal of "/ipfs/..." from multiaddrs
-      // this will be solved in: https://github.com/libp2p/js-libp2p/issues/323
-      this.peerInfo.multiaddrs.toArray().forEach(m => {
-        let ma
-        try {
-          ma = m.decapsulate('ipfs')
-        } catch (_) {
-          ma = m
-        }
-
-        this.peerInfo.multiaddrs.replace(m, ma)
-      })
-
-      callback()
-    })
-  }
-
-  /**
-   * Dials the given peer on protocol. The promise will resolve with the connection
-   * NOTE: This is currently promisified internally by libp2p
-   *
-   * @param {PeerInfo} peerInfo
-   * @param {string} protocol
-   * @param {function(Error, Connection)} callback
+   * @override
+   * @returns {Promise<void>}
    */
-  dial (peerInfo, protocol, callback) {
-    this.dialProtocol(peerInfo, protocol, (err, conn) => {
-      if (err) return callback(err)
-      if (!conn) return callback()
-
-      conn.getPeerInfo((err, peerInfo) => {
-        if (err) return callback(err)
-
-        // Convert the pull stream to an iterable node stream
-        const connection = pullToStream(conn)
-        connection.peerInfo = peerInfo
-
-        callback(null, connection)
+  async start () {
+    await super.start()
+    // replace with announce addrs until libp2p supports this directly
+    if (this.announceAddrs.length > 0) {
+      this.peerInfo.multiaddrs.clear()
+      this.announceAddrs.forEach(addr => {
+        this.peerInfo.multiaddrs.add(addr)
       })
-    })
-  }
+    }
 
-  /**
-   * Overrides the default `handle` to convert pull streams to streams
-   * NOTE: only convert if does not need pull-streams
-   * @param {string} protocol
-   * @param {function(Stream)} handler
-   */
-  handle (protocol, handler) {
-    super.handle(protocol, (_, conn) => {
-      if (this.needsPullStream) {
-        handler(protocol, conn)
-      } else {
-        conn.getPeerInfo((_, peerInfo) => {
-          const connection = pullToStream(conn)
-          connection.peerInfo = peerInfo
-          handler(protocol, connection)
-        })
+    // TODO: I think it can be removed
+    // temporary removal of "/ipfs/..." from multiaddrs
+    // this will be solved in: https://github.com/libp2p/js-libp2p/issues/323
+    this.peerInfo.multiaddrs.toArray().forEach(m => {
```

Comment:
can this be removed?

Reply:
I ran the tests without it and it worked fine. Unless it breaks interop we should be able to remove it

Reply:
We can try it. It **will** break interop if `ipfs` is on the end of a multiaddr. Go does not handle that case.